tensorstore/driver/tiff/compressor_registry.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TENSORSTORE_DRIVER_TIFF_COMPRESSOR_REGISTRY_H_
+#define TENSORSTORE_DRIVER_TIFF_COMPRESSOR_REGISTRY_H_
+
+#include <string_view>
+
+#include "absl/container/flat_hash_map.h"
+#include "tensorstore/internal/compression/json_specified_compressor.h"
+#include "tensorstore/internal/json_registry.h"
+#include "tensorstore/kvstore/tiff/tiff_details.h"
+
+namespace tensorstore {
+namespace internal_tiff {
+
+// Returns the global registry instance for TIFF compressors.
+// This registry maps string IDs (like "lzw", "deflate") to factories/binders
+// capable of creating JsonSpecifiedCompressor instances.
+internal::JsonSpecifiedCompressor::Registry& GetTiffCompressorRegistry();
+
```

Comment:
It seems unlikely that we need to register tiff compressor to start with, also the tiff tag value could map directly to a compressor instance (via an enum, or similar).